src/splinterdb.c

```diff
@@ -227,38 +229,95 @@ splinterdb_init_config(const splinterdb_config *kvs_cfg, // IN
  * Internal function for create or open
  */
 int
-splinterdb_create_or_open(const splinterdb_config *kvs_cfg,      // IN
-                          splinterdb             **kvs_out,      // OUT
-                          bool                     open_existing // IN
+splinterdb_create_or_open(splinterdb_config *kvs_cfg,      // IN
+                          splinterdb       **kvs_out,      // OUT
+                          bool               open_existing // IN
 )
 {
-   splinterdb     *kvs;
+   splinterdb     *kvs = NULL;
    platform_status status;
 
+   bool we_created_heap = FALSE;
+
+   // Allocate a shared segment if so requested. For now, we hard-code
+   // the required size big enough to run most tests. Eventually this
+   // has to be calculated here based on other run-time params.
+   // (Some tests externally create the platform_heap, so we should
+   // only create one if it does not already exist.)
+   if (kvs_cfg->use_shmem && (kvs_cfg->heap_handle == NULL)) {
+      size_t shmem_size = (kvs_cfg->shmem_size ? kvs_cfg->shmem_size : 2 * GiB);
+      status            = platform_heap_create(platform_get_module_id(),
```

Comment:
Attn: @rtjohnso - In Oct or so of 2022, when I first aired out the proto-version PR of this work, you had some major concern about this block of code. I didn't register it then as that airing was only a preview and wasn't quite ready for a formal review.

I forget now what your issue was about this chunk of code.

I am calling it out now, in case it jogs your memory as to what about this code piece you wanted a rework.

Let's discuss. ...